visitors/http_rest_diff/data_comparator.go

```diff
@@ -0,0 +1,27 @@
+package http_rest_diff
+
+import (
+	pb "github.com/akitasoftware/akita-ir/go/api_spec"
+	. "github.com/akitasoftware/akita-libs/visitors"
+	"github.com/akitasoftware/akita-libs/visitors/http_rest"
+)
+
+// Determines whether diffing between two Data instances will produce an empty
+// result.
+func IsSameData(d1, d2 *pb.Data) bool {
+	var comparator dataComparator
+	http_rest.ApplyPair(&comparator, d1, d2)
+	return !comparator.isDifferent
+}
+
+// A diff visitor for determining whether two Data instances are different.
+type dataComparator struct {
+	DefaultHttpRestSpecDiffVisitorImpl
+
+	isDifferent bool
+}
+
+func (v *dataComparator) EnterDiff(self interface{}, ctx http_rest.HttpRestSpecPairVisitorContext, left, right interface{}) Cont {
+	v.isDifferent = true
```

Comment:
I don't understand, why are two Data instances always different?

(Or, they're different only if they get to this case, DefaultHttpRestSpecDiffVisitorImpl covers everything else?)

Reply:
This is only ever called on two `Data` instances that have a diff—either `left` was removed, `right` was added, or `left` was changed to `right`. (Elsewhere, `left` is called `baseNode` and `right` is called `newNode`. I'll fix these parameter names to be consistent.)